### PR TITLE
3 fix ups for CATCloud

### PR DESCRIPTION
**Pull request by songlkkevin**
Fix three bugs in CATCloud.

### CODE REVIEW

**Review by pratiksampat — CHANGES_REQUESTED** [review]
s/substract/subtract

File: kernel/sched/core.c
Comment (pratiksampat) [inline kernel/sched/core.c:1]: Why is this line removed from here and added in the 3rd patch?

File: kernel/sched/core.c [hunk 1]
```diff
@@ -6,7 +6,6 @@
  *
  *  Copyright (C) 1991-2002  Linus Torvalds
  */
-#include "linux/rcupdate.h"
```
Comment (pratiksampat) [inline kernel/sched/core.c]: Why is this line removed from here and added in the 3rd patch?

File: kernel/sched/fair.c [hunk 2]
```diff
@@ -5486,12 +5486,12 @@ static int __assign_cfs_rq_runtime(struct cfs_bandwidth *cfs_b,
...
 		/* Find 99P of yeild and runtime for each runqueue maybe add it back to the queue itself */
-		if (temp_cfs_rq->pa_hist_idx - 1 <= 0)
+		if (temp_cfs_rq->pa_hist_idx<= 0)
 			continue;
-		sort(temp_cfs_rq->pa_yield_time_hist, temp_cfs_rq->pa_hist_idx - 1, sizeof(u64), cmp_u64, NULL);
-		sort(temp_cfs_rq->pa_runtime_hist, temp_cfs_rq->pa_hist_idx - 1, sizeof(u64), cmp_u64, NULL);
+		sort(temp_cfs_rq->pa_yield_time_hist, temp_cfs_rq->pa_hist_idx, sizeof(u64), cmp_u64, NULL);
+		sort(temp_cfs_rq->pa_runtime_hist, temp_cfs_rq->pa_hist_idx, sizeof(u64), cmp_u64, NULL);
 
-		percentile_idx = DIV_ROUND_UP(99 * (temp_cfs_rq->pa_hist_idx - 2), 100);
```
Comment (pratiksampat) [inline kernel/sched/fair.c:5493]: -2 here is intentional. P99 wants to remove outliers. for smaller history values, since we round up - P99 will always be the last entry and if there are outliers. On larger history values this also works because we have enough entries for P99 to work anyways.

**Review by pratiksampat — CHANGES_REQUESTED** [review]

File: kernel/sched/core.c [hunk 3]
```diff
@@ -6,6 +6,7 @@
  *
  *  Copyright (C) 1991-2002  Linus Torvalds
  */
+#include "linux/rcupdate.h"
```
Comment (pratiksampat) [inline kernel/sched/core.c]: Removed from patch 1 and added in patch 3. Otherwise the patchset LGTM.

**Review by pratiksampat — CHANGES_REQUESTED** [review]

File: kernel/sched/core.c (hunk 3 above)
Comment (songlkkevin) [reply kernel/sched/core.c]: This line is not necessary so I removed it in the last commit.

Comment (pratiksampat) [reply kernel/sched/core.c]: Why was is it added now? If its dead-code, then you can remove it from here as well and have a new commit with that!
